modules/core/src/main/java/org/locationtech/jts/algorithm/distance/DistanceFunction.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright (c) 2021 Felix Obermaier.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * and Eclipse Distribution License v. 1.0 which accompanies this distribution.
+ * The Eclipse Public License is available at http://www.eclipse.org/legal/epl-v20.html
+ * and the Eclipse Distribution License is available at
+ *
+ * http://www.eclipse.org/org/documents/edl-v10.php.
+ */
+package org.locationtech.jts.algorithm.distance;
+
+import org.locationtech.jts.geom.Coordinate;
+
+/**
+ * Base interface for classes that can compute a distance between two {@link Coordinate}s.
+ *
+ * @author Felix Obermaier
+ */
+public interface DistanceFunction {
```

Comment:
The term "function" is fairly overloaded in the JTS codebase.  Can this be changed to `DistanceMetric`?